domain/src/Aggregates/NonFungibleAsset/NonFungibleAsset.php

```diff
@@ -29,6 +33,8 @@ class NonFungibleAsset implements AggregateRoot
     /** @phpstan-use AggregateRootBehaviour<NonFungibleAssetId> */
     use AggregateRootBehaviour;
 
+    private ?Asset $asset = null;
```

Comment:
ℹ️ Made sense to save the asset at the aggregate level – makes some invariant more explicit